pkg/controller/worker/add.go

```diff
@@ -48,8 +54,29 @@ func AddToManagerWithOptions(mgr manager.Manager, opts AddOptions) error {
 		return err
 	}
 
+	delegateFactory := &delegateFactory{
+		logger: workerLogger,
+	}
```

Comment:
You could inject the `dataVolumeManager` into the `delegateFactory` here (see the comment above).